python/private/python.bzl

```diff
@@ -355,21 +438,146 @@ interpreter to create pyc files for the standard library, but, because they are
 created as needed, it adversely affects Bazel's ability to cache the runtime and
 can result in spurious build failures.
 """,
-                    mandatory = False,
-                ),
-                "is_default": attr.bool(
-                    mandatory = False,
-                    doc = "Whether the toolchain is the default version",
-                ),
-                "python_version": attr.string(
-                    mandatory = True,
-                    doc = "The Python version, in `major.minor` format, e.g " +
-                          "'3.12', to create a toolchain for. Patch level " +
-                          "granularity (e.g. '3.12.1') is not supported.",
-                ),
-            },
+            mandatory = False,
+        ),
+        "is_default": attr.bool(
+            mandatory = False,
+            doc = "Whether the toolchain is the default version",
+        ),
+        "python_version": attr.string(
+            mandatory = True,
+            doc = "The Python version, in `major.minor` format, e.g " +
+                  "'3.12', to create a toolchain for. Patch level " +
+                  "granularity (e.g. '3.12.1') is not supported.",
+        ),
+    },
+)
+
+_override = tag_class(
+    doc = """Tag class used to override defaults and behaviour of the module extension.
+
+:::{versionadded} 0.36.0
+:::
+""",
+    attrs = {
+        "available_python_versions": attr.string_list(
+            mandatory = False,
+            doc = "The list of available python tool versions to use. Must be in `X.Y.Z` format.",
```

Comment:
doc notes:
* What happens if an unknown version is given?
* Also mention the purpose for specifying this -- to prevent transitive dependencies from introducing additional overhead